pkg/locking/locking.go

```diff
@@ -74,13 +77,26 @@ func reInit() {
 	godeadlock.Opts.DeadlockTimeout = time.Duration(timeoutSec) * time.Second
 	godeadlock.Opts.LogBuf = &errorBuf{}
 	godeadlock.Opts.OnPotentialDeadlock = onPotentialDeadlock
+	if exitEnv, err := strconv.ParseBool(os.Getenv(EnvExitOnDeadlock)); err != nil {
+		exitOnDeadlock = false
+	} else {
+		exitOnDeadlock = exitEnv
+	}
+
 	if enabled {
-		fmt.Fprintf(os.Stderr, "=== Deadlock detection enabled (timeout: %d seconds) ===\n", timeoutSec)
+		fmt.Fprintf(os.Stderr, "=== Deadlock detection enabled (timeout: %d seconds, exit on deadlock: %v) ===\n", timeoutSec, exitOnDeadlock)
```

Comment:
this is not related to this PR, but why we don't use `log.Log(log.Diagnostics).Error`?

Reply:
That's intentional to avoid pulling in unwanted dependencies. We want to ensure that we write this before any other subsystem is initialized, including logging (which may also use locks). It's also important enough to not risk being suppressed.

Reply:
Added comment